gslib/tests/testcase/integration_testcase.py

```diff
@@ -1135,11 +1136,23 @@ def SetAnonymousBotoCreds(self):
                                    boto.config.get('GSUtil', api_config_key,
                                                    None)))
 
-    with SetBotoConfigForTest(boto_config_for_test, use_existing_config=False):
-      # Make sure to reset Developer Shell credential port so that the child
-      # gsutil process is really anonymous.
-      with SetEnvironmentForTest({'DEVSHELL_CLIENT_PORT': None}):
-        yield
+    # Prevent Boto from falling back on credentials from other files like
+    # ~/.aws/credentials or environment variables.
+    saved_aws_secrety_access_key = None
+    if AWS_SECRET_ACCESS_KEY in os.environ:
+      saved_aws_secrety_access_key = os.environ[AWS_SECRET_ACCESS_KEY]
+    os.environ[AWS_SECRET_ACCESS_KEY] = '_'
+
+    try:
+      with SetBotoConfigForTest(boto_config_for_test,
+                                use_existing_config=False):
+        # Make sure to reset Developer Shell credential port so that the child
+        # gsutil process is really anonymous.
+        with SetEnvironmentForTest({'DEVSHELL_CLIENT_PORT': None}):
```

Comment:
Isn't this function setting the envs for you? Can we just pass the values to this function?

Reply:
Wow, I didn't see that